### ISSUE TITLE
pylint incorrectly complains about missing member

### ISSUE THREAD

**Issue by frerich** [PyCQA/pylint#1004]
### Steps to reproduce
1. Create a Python script containing the lines

    ```python
    import sys
    sys.stdout.buffer.write(b'hey\n')
    ```

2. Execute aforementioned Python script via Python 3.5 (maybe any Python 3.x will do?). It should merely print `hey` and not run into any problems.
3. Run `pylint <scriptfile.py>`

### Current behavior
Pylint complains with
```
E:  2, 0: Class 'buffer' has no 'write' member (no-member)
```

### Expected behavior
Pylint should not produce any output at all.

### Workarounds
* The `buffer` field can be accessed via `getattr(sys.stdout, 'buffer')`.
* The error can be disabled locally by adding the inline option `# pylint: disable=no-member`.

### pylint --version output
pylint 1.5.6,
astroid 1.4.6
Python 3.5.1 (v3.5.1:37a07cee5969, Dec  6 2015, 01:54:25) [MSC v.1900 64 bit (AMD64)]

**Comment by The-Compiler** [comment]
I can reproduce this with pylint/astroid from git as well

**Comment by PCManticore** [comment]
I can reproduce it as well.

**Comment by glennmatthews** [comment]
Possibly related to PyCQA/astroid#347 ?

**Comment by PCManticore** [comment]
Probably not, this issue doesn't seem related to context managers.

**Comment by dargueta-work** [comment]
Has this been fixed? I see the issue is closed, but it's still broken for me in pylint 1.6.5, astroid 1.4.9 on Python 3.6.0.

**Comment by rogalski** [comment]
@dargueta-work it's fixed (7cf531c69dc495419ea5e5d3353c43fbc5b7b188) but not yet shipped. You have to wait for 1.7.0.

**Comment by dargueta-work** [comment]
Okay, great. Thanks!

**Comment by joshveea** [comment]
I'm still seeing this issue. 

Astroid 2.5.6
Pylint 2.8.0

**Comment by Pierre-Sassoulas** [comment]
@joshveea could you open a new issue for it please ?
